pkgs/by-name/li/liana/default.nix

```diff
@@ -61,27 +63,29 @@ rustPlatform.buildRustPackage rec {
     systemd
   ];
 
-  sourceRoot = "${src.name}/gui";
+  sourceRoot = if withGui then "${src.name}/gui" else null;
```

Comment:
Would the `sourceRoot` not be `"${src.name}"` for lianad instead of null?

Reply:
according to the [docs on `rustPlatform`](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#compiling-non-rust-packages-that-include-rust-code-compiling-non-rust-packages-that-include-rust-code):

> sourceRoot: when the `Cargo.lock`/`Cargo.toml` are in a subdirectory, `sourceRoot` specifies the relative path to these files.

so the rationale here was: if `withGui` is `false`, then we avoid defining `sourceRoot`

if I change this line to:
```
  sourceRoot = if withGui then "${src.name}/gui" else "${src.name}";
```

compilation also works, so I think either way should be fine.

Reply:
That's fine if it's default at root. Can leave it null.